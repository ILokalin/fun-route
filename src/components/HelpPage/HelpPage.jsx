import React from 'react';
import HelpCard from 'components/HelpCard';

import SetNewPointMapImage from 'assets/set-new-point-map.jpg';
import AddNewPointButtonImage from 'assets/add-new-point-button.jpg';
import StartEndPointsInRoutImage from 'assets/start-end-points-in-route.jpg';
import PlacemarkAddressRouteImage from 'assets/placemark-address-route.jpg';
import ButtonsUpDownImage from 'assets/buttons-up-down.jpg';
import RouteTypeButton from 'assets/route-type-button.jpg';
import ButtonRemoveImage from 'assets/button-remove.jpg';

import IconRemove from 'components/IconsBtn/IconRemove';
import IconAdd from 'components/IconsBtn/IconAdd';


export default class extends React.Component {

  render () {

    const help = [
      {
        title: 'Выбор точки',
        image: SetNewPointMapImage,
        text: ["Для начала работы необходимо выбрать место начала ма\u00adршрута на карте и уста\u00adновить маркер. Маркер можно установить пере\u00adтаски\u00adва\u00adнием маркера из центра карты или кликнув в нужной точке ка\u00adрты левой клавишей мыши."]
      }, {
        title: 'Добавление точки',
        image: AddNewPointButtonImage,
        text: ['Адрес отобразится в поле создания новой точки маршрута. В поле можно указать свое название для точки (адрес сохра\u00adнится и будет также указан), уточнить координаты. Для добавления точки необходимо нажать кнопку', IconAdd(), 'или клавишу Enter в поле для ввода текста.']
      }, {
        title: 'Список маршрута',
        image: StartEndPointsInRoutImage,
        text: ['Добавляя точки, их описание (адрес, название) разме\u00adщаются в списке.  Первые 10 получают литер от A до J. В случае, когда название или адрес не помещаются в ширину отведенного поля, можно получить разве\u00adрнутую информацию о точке маршрута, выделив точку левой кнопкой мыши.']
      }, {
        title: 'Маркеры на карте',
        image: PlacemarkAddressRouteImage,
        text: ['Для каждой точки на карте создается круглый маркер, место\u00adположение соответствует координатам (широта, долгота). Нажав на маркер, откры\u00adвается текущий адрес в соответствии с гео\u00adкодиро\u00adванием сервиса Яндекс-карты. Добавьте начальную и конечную точку маршрута в списке.']
      }, {
        title: 'Изменение последовательности',
        image: ButtonsUpDownImage,
        text: ['Для изменения последовательности маршрута можно пере\u00adта\u00adскивать элементы в списке, размещая их нужном порядке. Также, можно воспользоваться кнопками, которые открываются при наведении указателя мыши на элемент. В мобильной версии кнопки для пере\u00adмещения вверх/вниз открыты всегда, а литеры точек, в списке не указываются.']
      }, {
        title: 'Удаление точки',
        image: ButtonRemoveImage,
        text: ['Удалить точку из маршрута можно нажатием кнопки', IconRemove(), 'нахо\u00adдя\u00adщейся у каждого элемента в списке. Отменить удаление нево\u00adзможно.']
      }, {
        title: 'Режимы отображения маршрута',
        image: RouteTypeButton,
        text: ['Составленный маршрут можно представить в виде проло\u00adженного пути. Для этого необходимо переключить режим, выбрав “Показать маршрут на карте”. Вернуться в режим редактирования, можно установив переключатель в обра\u00adтное положение.']
      }]

    return (
      <section className="help">
        <h2 className="help__title-page">Описание работы</h2>
        
        <div className="help__rules-container">
          {help.map((element, index) => <HelpCard 
                                      key = {index.toString()}
                                      title = {element.title}
                                      image = {element.image}
                                      text = {element.text}
                                      />
          )}
  
        </div>
      </section>
  )

  } 
}
  