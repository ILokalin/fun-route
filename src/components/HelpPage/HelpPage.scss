.help {
  margin: 0 auto;
  max-width: 72.5rem;
  color: $gray-lightnes;
  
  box-sizing: border-box;
  padding-right: $side-gap;
  padding-left: $side-gap;

  
  &__title-page {
    padding-bottom: 1.875rem;
    text-align: center;
  }
  

  &__rules-container {
    display: grid;
    grid-template-columns: minmax(18.125rem, 1fr);
    grid-gap: 3.125rem;
    width: 100%;

    @media (min-width: $screen-sd) {
      grid-template-columns: minmax(23.125rem, 1fr);
    }

    @media (min-width: $screen-md) {
      grid-template-columns: repeat(auto-fill, minmax(25rem, 1fr)); 
    }

    @media (min-width: $screen-max) {
      padding: 0;
    }
  }
}
