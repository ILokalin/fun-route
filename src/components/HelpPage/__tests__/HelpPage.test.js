import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import rendrer from 'react-test-renderer';

import HelpPage from 'components/HelpPage';


describe('HelpPage Component', () => {
  it('render snap', () => {
    const component = rendrer.create(<HelpPage />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    })
})