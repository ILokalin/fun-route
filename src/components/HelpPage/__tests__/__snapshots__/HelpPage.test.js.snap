// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit render snap 1`] = `
<section
  className="help"
>
  <h2
    className="help__title-page"
  >
    Описание работы
  </h2>
  <div
    className="help__rules-container"
  >
    <div
      className="help-card"
    >
      <h3
        className="help-card__title"
      >
        Выбор точки
      </h3>
      <div
        className="help-card__image-section"
      >
        <img
          alt="Выбор точки"
          className="help-card__image"
          src="set-new-point-map.jpg"
        />
      </div>
      <p
        align="justify"
        className="help-card__text"
      >
        Для начала работы необходимо выбрать место начала ма­ршрута на карте и уста­новить маркер. Маркер можно установить пере­таски­ва­нием маркера из центра карты или кликнув в нужной точке ка­рты левой клавишей мыши.
      </p>
    </div>
    <div
      className="help-card"
    >
      <h3
        className="help-card__title"
      >
        Добавление точки
      </h3>
      <div
        className="help-card__image-section"
      >
        <img
          alt="Добавление точки"
          className="help-card__image"
          src="add-new-point-button.jpg"
        />
      </div>
      <p
        align="justify"
        className="help-card__text"
      >
        Адрес отобразится в поле создания новой точки маршрута. В поле можно указать свое название для точки (адрес сохра­нится и будет также указан), уточнить координаты. Для добавления точки необходимо нажать кнопку
        <svg
          aria-labelledby="title"
          className="icon"
          fill="none"
          viewBox="0 0 30 30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <title
            id="title"
          >
            Добавить точку в маршрут
          </title>
          <path
            className="icon__path"
            d="M13.893 8.555v5.274H8.619a1.172 1.172 0 100 2.344h5.274v5.33a1.172 1.172 0 102.343 0v-5.33h5.274a1.172 1.172 0 100-2.344h-5.274V8.555a1.172 1.172 0 10-2.343 0zm14.935.879c.647 0 1.172.524 1.172 1.171v14.707A4.693 4.693 0 0125.312 30H4.688A4.693 4.693 0 010 25.312V4.688A4.693 4.693 0 014.688 0h20.625A4.693 4.693 0 0130 4.688a1.172 1.172 0 11-2.344 0 2.346 2.346 0 00-2.343-2.344H4.688a2.346 2.346 0 00-2.344 2.344v20.625a2.346 2.346 0 002.344 2.343h20.625a2.346 2.346 0 002.343-2.343V10.605c0-.648.525-1.172 1.172-1.172z"
            fill="#E9E9E9"
          />
        </svg>
        или клавишу Enter в поле для ввода текста.
      </p>
    </div>
    <div
      className="help-card"
    >
      <h3
        className="help-card__title"
      >
        Список маршрута
      </h3>
      <div
        className="help-card__image-section"
      >
        <img
          alt="Список маршрута"
          className="help-card__image"
          src="start-end-points-in-route.jpg"
        />
      </div>
      <p
        align="justify"
        className="help-card__text"
      >
        Добавляя точки, их описание (адрес, название) разме­щаются в списке.  Первые 10 получают литер от A до J. В случае, когда название или адрес не помещаются в ширину отведенного поля, можно получить разве­рнутую информацию о точке маршрута, выделив точку левой кнопкой мыши.
      </p>
    </div>
    <div
      className="help-card"
    >
      <h3
        className="help-card__title"
      >
        Маркеры на карте
      </h3>
      <div
        className="help-card__image-section"
      >
        <img
          alt="Маркеры на карте"
          className="help-card__image"
          src="placemark-address-route.jpg"
        />
      </div>
      <p
        align="justify"
        className="help-card__text"
      >
        Для каждой точки на карте создается круглый маркер, место­положение соответствует координатам (широта, долгота). Нажав на маркер, откры­вается текущий адрес в соответствии с гео­кодиро­ванием сервиса Яндекс-карты. Добавьте начальную и конечную точку маршрута в списке.
      </p>
    </div>
    <div
      className="help-card"
    >
      <h3
        className="help-card__title"
      >
        Изменение последовательности
      </h3>
      <div
        className="help-card__image-section"
      >
        <img
          alt="Изменение последовательности"
          className="help-card__image"
          src="buttons-up-down.jpg"
        />
      </div>
      <p
        align="justify"
        className="help-card__text"
      >
        Для изменения последовательности маршрута можно пере­та­скивать элементы в списке, размещая их нужном порядке. Также, можно воспользоваться кнопками, которые открываются при наведении указателя мыши на элемент. В мобильной версии кнопки для пере­мещения вверх/вниз открыты всегда, а литеры точек, в списке не указываются.
      </p>
    </div>
    <div
      className="help-card"
    >
      <h3
        className="help-card__title"
      >
        Удаление точки
      </h3>
      <div
        className="help-card__image-section"
      >
        <img
          alt="Удаление точки"
          className="help-card__image"
          src="button-remove.jpg"
        />
      </div>
      <p
        align="justify"
        className="help-card__text"
      >
        Удалить точку из маршрута можно нажатием кнопки
        <svg
          className="icon"
          fill="none"
          viewBox="0 0 30 30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <title
            id="title"
          >
            Убрать точку маршрута
          </title>
          <g
            className="icon__path"
            clipPath="url(#clip0)"
            fill="#798995"
          >
            <path
              d="M19.143 10.857c.458.457.458 1.2 0 1.657L16.657 15l2.486 2.486a1.172 1.172 0 11-1.657 1.657L15 16.657l-2.486 2.486a1.172 1.172 0 01-1.657-1.657L13.343 15l-2.486-2.486a1.172 1.172 0 111.657-1.657L15 13.343l2.486-2.486a1.172 1.172 0 011.657 0zM20.602 3.621l-1.516-.628a1.172 1.172 0 01.897-2.165l1.515.628a1.172 1.172 0 11-.896 2.165z"
            />
            <path
              d="M15 0a1.172 1.172 0 110 2.344C8.005 2.344 2.344 8.004 2.344 15c0 6.995 5.66 12.656 12.656 12.656 6.995 0 12.656-5.66 12.656-12.656A1.172 1.172 0 0130 15c0 8.29-6.709 15-15 15-8.29 0-15-6.709-15-15C0 6.71 6.709 0 15 0zM24.53 6.63l-1.16-1.16a1.172 1.172 0 111.656-1.657l1.16 1.16a1.172 1.172 0 11-1.657 1.658zM28.538 11.549a1.172 1.172 0 01-1.531-.634l-.628-1.516a1.172 1.172 0 112.165-.897l.628 1.516a1.172 1.172 0 01-.634 1.53z"
            />
          </g>
          <defs>
            <clipPath
              id="clip0"
            >
              <path
                d="M0 0h30v30H0z"
                fill="#fff"
                transform="matrix(-1 0 0 1 30 0)"
              />
            </clipPath>
          </defs>
        </svg>
        нахо­дя­щейся у каждого элемента в списке. Отменить удаление нево­зможно.
      </p>
    </div>
    <div
      className="help-card"
    >
      <h3
        className="help-card__title"
      >
        Режимы отображения маршрута
      </h3>
      <div
        className="help-card__image-section"
      >
        <img
          alt="Режимы отображения маршрута"
          className="help-card__image"
          src="route-type-button.jpg"
        />
      </div>
      <p
        align="justify"
        className="help-card__text"
      >
        Составленный маршрут можно представить в виде проло­женного пути. Для этого необходимо переключить режим, выбрав “Показать маршрут на карте”. Вернуться в режим редактирования, можно установив переключатель в обра­тное положение.
      </p>
    </div>
  </div>
</section>
`;
